### PR TITLE
Add more detail to errors

### PR DESCRIPTION
**Pull request by stringhandler**
Add more details to errors for better diagnosis of failures

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ repository = "https://github.com/tari-project/randomx-rs"
 homepage = "https://tari.com"
 readme = "README.md"
 license = "BSD-3-Clause"
-version = "0.4.1"
+version = "0.4.2"
```
Comment (StriderDM) [inline Cargo.toml]: Think this should be 0.5 since it a breaking change for existing errors.

**Review by delta1 — DISMISSED** [review]
LGTM - untested

File: Cargo.toml (hunk 1 above)
Comment (stringhandler) [reply Cargo.toml]: The public interface didn't change though

File: src/lib.rs [hunk 2]
```diff
@@ -295,11 +306,15 @@ impl RandomXVM {
...
-            return Err(RandomXError::FlagConfigError);
+            return Err(RandomXError::FlagConfigError(
+                "No cache and FLAG_FULL_MEM not set".to_string(),
+            ));
         }
 
         if dataset.is_none() && is_full_mem {
-            return Err(RandomXError::FlagConfigError);
+            return Err(RandomXError::FlagConfigError(
+                "No dataset and FLAG_FULL_MEM not set".to_string(),
```
Comment (StriderDM) [inline src/lib.rs]:
```
 "No dataset and FLAG_FULL_MEM is set".to_string(),
 ```

File: src/lib.rs [hunk 3]
```diff
@@ -320,7 +335,7 @@ impl RandomXVM {
         }
 
         if test.is_null() {
-            return Err(RandomXError::CreationError);
+            return Err(RandomXError::CreationError("VM not created".to_string()));
```
Comment (StriderDM) [inline src/lib.rs]:
```
return Err(RandomXError::CreationError("Failure to allocate VM".to_string()));
```

**Comment by stringhandler** [comment]
@StriderDM fixed issues and bumped version

**Review by StriderDM — APPROVED** [review]
LGTM

**Pull request merged**